File: ggpy/cruft/autocode/FileUtils.py
#!/usr/bin/env python
""" generated source for module FileUtils """
# package: org.ggp.base.util.files
import java.io.BufferedReader

import java.io.File

import java.io.FileNotFoundException

import java.io.FileOutputStream

import java.io.FileReader

import java.io.IOException

import java.io.PrintStream

class FileUtils(object):
    """ generated source for class FileUtils """
    # 
    #      * @param filePath the name of the file to open.
    #      
    @classmethod
    def readFileAsString(cls, file_):
        """ generated source for method readFileAsString """
        try:
            while (numRead = reader.read(buf)) != -1:
                fileData.append(buf, 0, numRead)
            reader.close()
            return fileData.__str__()
        except FileNotFoundException as e:
            return None
        except IOException as e:
            e.printStackTrace()
            return None

    @classmethod
    def writeStringToFile(cls, file_, s):
        """ generated source for method writeStringToFile """
        out = PrintStream(FileOutputStream(file_, False))
        try:
            out.print_(s)
        finally:
            out.close()

File: ggp-base/src/org/ggp/base/util/files/FileUtils.java
package org.ggp.base.util.files

import java.io.BufferedReader
import java.io.File
import java.io.FileNotFoundException
import java.io.FileOutputStream
import java.io.FileReader
import java.io.IOException
import java.io.PrintStream

class FileUtils(object):
    /**
     * @param filePath the name of the file to open.
     */
    def String readFileAsString(File file):
        try 
        	BufferedReader reader = new BufferedReader(new FileReader(file))
            StringBuilder fileData = new StringBuilder(10000)
            char[] buf = new char[1024]
            int numRead=0
            while((numRead=reader.read(buf)) != -1)
                fileData.append(buf, 0, numRead)

            reader.close()
            return fileData.toString()
        except FileNotFoundException e):
            return null
        except IOException e):
            e.printStackTrace()
            return null

    def void writeStringToFile(File file, String s) throws IOException 
        PrintStream out = new PrintStream(new FileOutputStream(file, false))
        try 
            out.print(s)
        } finally 
            out.close()

